cooked-validators/src/Cooked/MockChain/Monad/Direct.hs

```diff
@@ -246,13 +246,12 @@ utxoIndex0 = utxoIndex0From def
 -- ** Direct Interpretation of Operations
 
 instance (Monad m) => MonadBlockChain (MockChainT m) where
-  validateTxSkel skelUnbal = do
-    -- TODO We use the first signer as a default wallet for fees and
-    -- collaterals. These should become parameters of validateTxSkel in the
-    -- future
-    firstSigner NE.:| _ <- askSigners
-    let balancingWalletPkh = walletPKHash firstSigner
-    let collateralWallet = firstSigner
+  type BalanceAndCollateralSource (MockChainT m) = Wallet
+
+  validateTxSkel bcWallet moreSigners skelUnbal = do
+    let balancingWalletPkh = walletPKHash bcWallet
+        collateralWallet = bcWallet
+        allSigners = NE.nub (bcWallet NE.:| moreSigners) -- TODO also use skelUnbal's signers?
```

Comment:
The two indeed seem redundant (so is this review comment with my previous one). I think we need to choose only one way to specify additional signers. `as` and `SignedBy` have always been confusing for users of cooked so I expect this would be too.